Enable `ocamlformat` on the part of the codebase that doesn't use CPPO

Leonidas-from-XIV opened the pull request:
This enables `ocamlformat` on the subset of the codebase that does not use CPPO.

panglesd reviewed (approved):
I approve. I would also suggest adding a `.git-blame-ignore-revs` so that `git blame` output more meaningful result.

panglesd reviewed (changes requested):
In fact, currently not only `.cppo.ml` contains cppo instructions, see for instance `monomorphic.ml`.

They should either be renamed, or added to the `ocamlformat-ignore`!

Leonidas-from-XIV commented:
Oh yes, that's true, you're right. I somehow assumed only the `.cppo.` files had CPPO in them, so I added them to the ignore-file to avoid churn. Turns out `ocamlformat` didn't reformat them anyway (so I didn't have to undo anything), maybe it has gotten sufficiently smart to skip them. Still good to have it skip over.

Also, I added the revision of the commit to the file. Unfortunately, there's no default name that git would pick up and users need to configure git for this to work but adding it won't hurt.

panglesd reviewed (approved):
Looks good!

> Also, I added the revision of the commit to the file. Unfortunately, there's no default name that git would pick up and users need to configure git for this to work but adding it won't hurt.

I think `.git-blame-ignore-revs` is the name that github picks, so it's at least directly useful in the github interface. (https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

Leonidas-from-XIV commented:
Ah, that's great to know, definitely useful!
